### PR TITLE
avl_tree.cpp: Add AVL tree

### PR DESCRIPTION
**Pull request by paawan01**
This adds avl tree in cpp with search, insert and delete.

Closes https://github.com/NITSkmOS/Algorithms/issues/48

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ This repository contains examples of various algorithms written on different pro
...
 | [Stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type))                               | [:octocat:](stack/C )                 |                                       | [:octocat:](stack/Java)               | [:octocat:](stack/Python)             |
 | [Linear Linked List](https://en.wikipedia.org/wiki/Linked_list)                                 | [:octocat:](linked_list/C)            | [:octocat:](linked_list/Cpp)          |                                       | [:octocat:](linked_list/Python)       |
-| [AVL Tree](https://en.wikipedia.org/wiki/AVL_tree)                                              | [:octocat:](avl_tree/C)               |                                       | [:octocat:](avl_tree/Java)            | [:octocat:](avl_tree/Python)          |
+| [AVL Tree](https://en.wikipedia.org/wiki/AVL_tree)                                              | [:octocat:](avl_tree/C)               | [:octocat:](avl_tree/Cpp)                          | [:octocat:](avl_tree/Java)            | [:octocat:](avl_tree/Python)          |
```
Comment (sangamcse) [inline README.md]: Cpp column characters should be 40. It is more than 40 which is making it little odd.

File: avl_tree/Cpp/avl_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+		root = delete_node_util(root, v);
+	}
+
+	private:
+	node * root;
+
+	/*
+	Class' private member functions
+	*/
+	node * search_util(node * t, int v) {
```
Comment (sangamcse) [inline avl_tree/Cpp/avl_tree.cpp]: what is `t` and `v` is missing in the doc string

File: avl_tree/Cpp/avl_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+		if(t == NULL || t->get_val() == v) return t;
+		else if (v < t->get_val())
+			return search_util(t->get_left(), v);
+		else
+			return search_util(t->get_right(), v);
+	}
+
+	node * insert_util(node * t, int v) {
```
Comment (sangamcse) [inline avl_tree/Cpp/avl_tree.cpp]: same here

File: avl_tree/Cpp/avl_tree.cpp (hunk 2 above)
Comment (paawan01) [reply avl_tree/Cpp/avl_tree.cpp]: Done 👍

File: avl_tree/Cpp/avl_tree.cpp (hunk 3 above)
Comment (paawan01) [reply avl_tree/Cpp/avl_tree.cpp]: Done 👍

**Comment by sangamcse** [comment]
ack 67e303f

**Comment by sangamcse** [comment]
ack 67e303fb685131b91341c25fd17ebe88336fb13a

**Comment by sangamcse** [comment]
@sangamcse ff

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being fastforwarded automatically. Please **DO NOT** push while fastforward is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated fastforward with [GitMate.io](https://gitmate.io) was successful! :tada:

**Pull request merged**